### ISSUE TITLE
"Run every day at midnight" only runs every other day

### ISSUE THREAD

**Issue by tverghis** [mdsherry/clokwerk#22]
I have code like so:

```rust
fn main() {
    env_logger::init();

    let mut scheduler = Scheduler::new();

    scheduler
        .every(1.days())
        .at("00:00")
        .run(do_the_thing);

    loop {
        scheduler.run_pending();
        thread::sleep(Duration::from_millis(200));
    }
}
```

From the logs generated by my application `do_the_thing` is only called once every _other_ day at midnight. I would expect the function to be called every day at midnight.

`do_the_thing` is very lightweight; it pretty much just sends out 3 HTTP POST requests, and does no time-consuming processing.

Is this a bug, or expected behavior? If this is expected behavior, how do I set up the scheduler to achieve what I want? Thanks!

**Comment by mdsherry** [comment]
I'd say it's a bug because I suspect `scheduler.every(1.days()).at("00:01")` would work as expected. You can get your code working by just removing the `at` since midnight is the default time. I think the problem is that it's looking for the *next* midnight from that reference point, which would be the next night. Compare to any other time, where it would be later the same day.

**Comment by tverghis** [comment]
Thanks for confirming, and providing a solution! I'll try that out.

Might be me just being daft, but I guess it's not apparent to me why midnight would be a special case as you stated -- seems like it's just like any other time. If you have the time to expand on your theory, I'd love to understand it. If not, no worries :)

(Not closing the issue since you suspect this is an unaddressed bug, but I can work around it)

**Comment by mdsherry** [comment]
I've just published 0.3.5, which has a fix for this.
